docs/design/milestone_1.md

```diff
@@ -0,0 +1,335 @@
+
+# Criticality Score Revamp: Milestone 1
+
+- Author: [calebbrown@google.com](mailto:calebbrown@google.com)
+- Updated: 2022-04-29
+
+
+## Goal
+
+Anyone can reliably generate raw signal data using the `criticality_score`
+GitHub project.
+
+Support future moves towards scaling and automating criticality score.
+
+### Non-goals
+
+**Improve how the score is calculated.**
+
+While this is overall vital, the ability to calculate the score depends on
+having reliable signals to base the score on.
+
+**Cover source repositories hosted on non-GitHub hosts.**
+
+Critical projects are hosted on GitLab, Bitbucket, or even self-hosted. These
+should be supported, but given that over 90% of open source projects are
+hosted by GitHub it seems prudent to focus efforts there first.
+
+## Background
+
+TBC
+
+## Design Overview
+
+### Project Definition
```

Comment:
Might be good to move some of these definitions and discussions (projects/forks) to a glossary of sorts (either at the end of this document or in a separate one). 

I think the main points of this milestone doc should come across more clearly:

- this is rearchitecturing the project to enable more reliability and extensibility
- focusing on:
  - reliably enumerating GitHub projects
  - Reliably generating existing signals. Adding dependent information.
  - Making the current criticality rankings list generated more frequently. (And more easily)

Reply:
PTAL. I've tried to incorporate this